Yona/Yona/MeDashBoard/YonaUserProfileViewController.swift

```diff
@@ -276,43 +276,26 @@ class YonaUserProfileViewController: UIViewController, UITableViewDelegate, UITa
      }
     
     func isUserDataValid() -> validateError {
-        if aUser?.firstName.characters.count == 0 {
-            self.displayAlertMessage("", alertDescription:
-                NSLocalizedString("enter-first-name-validation", comment: ""))
+        if aUser?.firstName.count == 0 {
+            self.displayAlertMessage("", alertDescription:NSLocalizedString("enter-first-name-validation", comment: ""))
             return .firstname
-        }
-        else if aUser?.lastName.characters.count == 0 {
-            self.displayAlertMessage("", alertDescription:
-                NSLocalizedString("enter-last-name-validation", comment: ""))
+        } else if aUser?.lastName.count == 0 {
+            self.displayAlertMessage("", alertDescription: NSLocalizedString("enter-last-name-validation", comment: ""))
             return .lastname
-        } else if aUser?.mobileNumber.characters.count == 0 {
-            self.displayAlertMessage("", alertDescription:
-                NSLocalizedString("enter-number-validation", comment: ""))
+        } else if aUser?.mobileNumber.count == 0 {
+            self.displayAlertMessage("", alertDescription:NSLocalizedString("enter-number-validation", comment: ""))
             return .phone
-        } else if aUser?.nickname.characters.count == 0 {
-            self.displayAlertMessage("", alertDescription:
-                NSLocalizedString("enter-nickname-validation", comment: ""))
+        } else if aUser?.nickname.count == 0 {
+            self.displayAlertMessage("", alertDescription: NSLocalizedString("enter-nickname-validation", comment: ""))
             return .nickname
         } else {
-            var number = ""
             if let mobilenum = aUser?.mobileNumber {
-                number =  mobilenum
-                
-                let trimmedWhiteSpaceString = number.removeWhitespace()
-                let trimmedString = trimmedWhiteSpaceString.removeBrackets()
-                
-                if trimmedString.validateMobileNumber() == false {
-                    self.displayAlertMessage("", alertDescription:
-                        NSLocalizedString("enter-number-validation", comment: ""))
+                if mobilenum.removeWhitespace().removeBrackets().validateMobileNumber() == false {
```

Comment:
This isn't right. The original logic made a number of corrections to the user input:
* Remove white spaces
* Remove brackets
* Replace 310 with +31

The first two corrections were good. The intend of the last one was good, but it wasn't implemented well.

Your code now doesn't do any of the corrections anymore. Please bring the first to corrections back and implement the third one in a better way. See also [my comment on the ticket](https://jira.yona.nu/browse/APPDEV-1131?focusedCommentId=14613&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14613).

Reply:
I've updated the code accordingly. For third point, first it will check entered mobile num has prefix (310) or not, if yes then replace with (+31).